pkg/migration-tool/writer/writer.go

```diff
@@ -24,22 +20,26 @@ const (
 type RemoteWrite struct {
 	c                  context.Context
 	sigBlockRead       chan *planner.Block
-	client             *utils.Client
+	url                string
+	shardsSet          *shardsSet
+	numShards          int
 	blocksPushed       atomic.Int64
 	progressMetricName string // Metric name to main the last pushed maxt to remote write storage.
 	migrationJobName   string // Label value to the progress metric.
 	progressTimeSeries *prompb.TimeSeries
 }
 
 // New returns a new remote write. It is responsible for writing to the remote write storage.
-func New(c context.Context, remoteWriteUrl, progressMetricName, migrationJobName string, sigRead chan *planner.Block) (*RemoteWrite, error) {
-	wc, err := utils.NewClient(fmt.Sprintf("writer-%d", 1), remoteWriteUrl, utils.Write, model.Duration(defaultWriteTimeout))
+func New(c context.Context, remoteWriteUrl, progressMetricName, migrationJobName string, numShards int, sigRead chan *planner.Block) (*RemoteWrite, error) {
+	ss, err := newShardsSet(c, remoteWriteUrl, numShards)
 	if err != nil {
-		return nil, fmt.Errorf("creating write-client: %w", err)
+		return nil, fmt.Errorf("creating shards: %w", err)
 	}
 	write := &RemoteWrite{
 		c:                  c,
-		client:             wc,
+		shardsSet:          ss,
+		url:                remoteWriteUrl,
+		numShards:          numShards,
```

Comment:
is numShards required if we have access to shardsSet.num?

Reply:
Yeah, we do not require `numShards` anymore. Thank you for pointing it out.